compiler/rustc_codegen_llvm/src/base.rs

```diff
@@ -93,6 +110,45 @@ pub fn iter_globals(llmod: &'ll llvm::Module) -> ValueIter<'ll> {
     unsafe { ValueIter { cur: llvm::LLVMGetFirstGlobal(llmod), step: llvm::LLVMGetNextGlobal } }
 }
 
+pub fn svh_symbol_name(
+    tcx: TyCtxt<'_>,
+    svh: &Svh, //, cgu_name: Symbol
+) -> String {
+    format!("rust_svh_{}_{}", tcx.original_crate_name(LOCAL_CRATE), svh)
+}
+
+// cx: &mut CodegenCx<'b, '_>,
+fn add_svh(tcx: TyCtxt<'tcx>, llvm_module: &'ll ModuleLlvm, svh: &Svh) -> &'ll Value
+//where
+//    'tcx: 'b,
+{
+    // Add SVH @todo insert symbol here
+    let (metadata_llcx, metadata_llmod) = (&*llvm_module.llcx, llvm_module.llmod());
+
+    let svh_bytes: Vec<u8> = format!("{}", svh).as_bytes().to_vec();
+    let llconst = common::bytes_in_context(metadata_llcx, &svh_bytes);
+    let name = svh_symbol_name(tcx, &svh); //cgu_name
+    let buf = CString::new(name).unwrap();
+    let llglobal =
+        unsafe { llvm::LLVMAddGlobal(metadata_llmod, common::val_ty(llconst), buf.as_ptr()) };
+    unsafe {
+        llvm::LLVMSetInitializer(llglobal, llconst);
+
+        let section_name = format!(
+            "__DATA,.rust_svh_hash", //TO DO make work for non-OSX not ,no_dead_strip
+        );
+        let name = SmallCStr::new(&section_name);
+        llvm::LLVMSetSection(llglobal, name.as_ptr());
+
+        // Also generate a .section directive to force no
+        // flags, at least for ELF outputs, so that the
+        // metadata doesn't get loaded into memory.
+        let directive = format!(".section {}", section_name);
+        llvm::LLVMSetModuleInlineAsm2(metadata_llmod, directive.as_ptr().cast(), directive.len())
```

Comment:
This overwrites the module inline asm instead of appending. You need to use `LLVMRustAppendModuleInlineAsm` if you want to append.